eras/shelley/impl/src/Cardano/Ledger/Shelley/UTxO.hs

```diff
@@ -67,11 +68,10 @@ scriptStakeCred ::
   ShelleyEraTxCert era =>
   TxCert era ->
   Maybe (ScriptHash (EraCrypto era))
-scriptStakeCred (ShelleyTxCertDeleg (DeRegKey (KeyHashObj _))) = Nothing
-scriptStakeCred (ShelleyTxCertDeleg (DeRegKey (ScriptHashObj hs))) = Just hs
-scriptStakeCred (ShelleyTxCertDeleg (Delegate (Delegation (KeyHashObj _) _))) = Nothing
-scriptStakeCred (ShelleyTxCertDeleg (Delegate (Delegation (ScriptHashObj hs) _))) = Just hs
-scriptStakeCred _ = Nothing
+scriptStakeCred = \case
+  ShelleyTxCertDeleg (ShelleyUnRegCert cred) -> scriptCred cred
+  ShelleyTxCertDeleg (ShelleyDelegCert cred _) -> scriptCred cred
+  _ -> Nothing
```

Comment:
Wouldn't it be better to explicitly mention the other case instead of the catchall, in case a new constructor will be added, so we have to explicitly define the behaviour for it?

Reply:
Those types are unlikely to change the number of constructors, but it doesn't hurt to do it anyhow. I changed it to:
```diff
-  ShelleyTxCertDeleg (ShelleyUnRegCert cred) -> scriptCred cred
-  ShelleyTxCertDeleg (ShelleyDelegCert cred _) -> scriptCred cred
+  ShelleyTxCertDeleg delegCert ->
+    case delegCert of
+      ShelleyRegCert _ -> Nothing
+      ShelleyUnRegCert cred -> scriptCred cred
+      ShelleyDelegCert cred _ -> scriptCred cred
```